google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java

```diff
@@ -162,7 +166,11 @@ public Query range(ByteStringRange range) {
    * filters, please use {@link Filters#interleave()} or {@link Filters#chain()}.
    */
   public Query filter(Filters.Filter filter) {
-    builder.setFilter(filter.toProto());
+    Preconditions.checkNotNull(filter, "filter can't be null");
```

Comment:
can we add a test please?

Reply:
Thanks for the review. Added test case for null check. I haven't added any test case for the filter size. I hope that should be fine.

Reply:
Apologies for being unclear, if you could add a test case for filter size as well that would be awesome :)

Reply:
I intentionally didn't add test case for that. Reason being the filter size needs to be more than 20MB, I thought that'd be a little large.

I would add unit test right away.

Reply:
Understood - LGTM. @igorbernstein2 any other thoughts?